src/nccl_ofi_sendrecv.c

```diff
@@ -1600,12 +1597,8 @@ static int listen(nccl_net_ofi_ep_t *base_ep,
 
 	/* Only 1 address should be inserted into the AV */
 	if (OFI_UNLIKELY(num_addrs != 1)) {
-		NCCL_OFI_WARN("Unable to insert remote address into address vector for device %d. RC: %s",
-			      dev_id, fi_strerror(-ret));
-		ret = -EINVAL;
-		goto error;
-	} else {
-		ret = 0;
+		NCCL_OFI_WARN("Unable to insert remote address into address vector for device %d.", dev_id);
+		return -EINVAL;
```

Comment:
Please make most of these changes.  I'm just going to have to undo them when we get rid of all the fine grained locking.

Reply:
I do not understand what this means.

Reply:
It means I can't type.  I meant to say "Please to not make these changes"

Reply:
What change specifically, and why would you have to undo it when you make locking changes?

The main reason this specific hunk exists is because we're calling fi_strerror on ret, but ret was never filled with any error code. That broken fi_strerror call has been here since March of 2023:  da33b97507bc5d18bd6a2b094f8380f3a54705b1

Reply:
We've talked a couple times that we need to get into Libfabric's FI_THREAD_DOMAIN model, which is most efficiently implemented (given how NCCL threading works) as a big kernel lock style implementation.  That would suggest that most of listen() and friends are under a single lock that is released at the end of the function, making removing all the exception handling escapes somethig I'm just going to go back and undo.

Reply:
I would probably just create a wrapper that acquires and releases the lock externally.